🐛 [Bug] 브랜치를 머지하면서 생긴 버그들을 수정합니다.

### Problem

브랜치를 머지하면서 빌드가 되지 않습니다.

### Reproduction

./gradlew build

### Screenshot

_No response_

### Expected Behavior

_No response_

### Log

```Shell
➜  ReBook_BE git:(develop) gradlew build  

> Task :rebook-product:compileJava FAILED
/Users/kimjaehyuk/Desktop/ReBook_BE/rebook-product/src/main/java/com/be/rebook/product/entity/Product.java:40: warning: @Builder will ignore the initializing expression entirely. If you want the initializing expression to serve as default, add @Builder.Default. If it is not supposed to be settable during building, make the field final.
    private List<ProductImage> productImages = new ArrayList<>();
                               ^
/Users/kimjaehyuk/Desktop/ReBook_BE/rebook-product/src/main/java/com/be/rebook/product/controller/ProductController.java:74: error: incompatible types: String cannot be converted to MemberLoginInfo
                .body(new BaseResponse<>(productService.findAllMyProducts(memberLoginInfo.getUsername())));
                                                                                                     ^
/Users/kimjaehyuk/Desktop/ReBook_BE/rebook-product/src/main/java/com/be/rebook/product/controller/ProductController.java:93: error: method changeStatus in class ProductService cannot be applied to given types;
                .body(new BaseResponse<>(productService.changeStatus(productId, productStatusRequestDTO.getStatus())));
                                                       ^
  required: MemberLoginInfo,Long,String
  found:    Long,String
  reason: actual and formal argument lists differ in length
/Users/kimjaehyuk/Desktop/ReBook_BE/rebook-product/src/main/java/com/be/rebook/product/controller/ProductController.java:93: error: incompatible types: Long cannot be converted to MemberLoginInfo
                .body(new BaseResponse<>(productService.changeStatus(productId, productStatusRequestDTO.getStatus())));
                                                                     ^
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
3 errors
1 warning

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':rebook-product:compileJava'.
> Compilation failed; see the compiler error output for details.

* Try:
> Run with --info option to get more log output.
> Run with --scan to get full insights.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.8/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 4s
18 actionable tasks: 1 executed, 17 up-to-date
➜  ReBook_BE git:(develop)
```


### Browser

_No response_

--- a/rebook-product/src/main/java/com/be/rebook/product/controller/ProductController.java
+++ b/rebook-product/src/main/java/com/be/rebook/product/controller/ProductController.java
@@ -4,12 +4,9 @@ import com.be.rebook.common.argumentresolver.auth.Auth;
 import com.be.rebook.common.argumentresolver.auth.MemberLoginInfo;
 import com.be.rebook.common.config.BaseResponse;
 import com.be.rebook.common.dto.PaginationResponseDTO;
-import com.be.rebook.common.exception.BaseException;
-import com.be.rebook.common.exception.ErrorCode;
 import com.be.rebook.product.service.ProductService;
 import jakarta.validation.Valid;
 import lombok.RequiredArgsConstructor;
-import org.springframework.data.domain.Page;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
@@ -49,10 +46,11 @@ public class ProductController {
      */
     @PutMapping("{productId}")
     public ResponseEntity<BaseResponse<Long>> updateProduct(@PathVariable("productId") Long productId,
-                                                            @Auth MemberLoginInfo memberLoginInfo,
-                                                            @RequestPart("productRequest") ProductUpdateRequestDTO productUpdateRequestDTO,
-                                                            @RequestPart("imageFiles") List<MultipartFile> imageFiles) throws IOException {
-        return ResponseEntity.ok().body(new BaseResponse<>(productService.updateProduct(productId, memberLoginInfo, productUpdateRequestDTO, imageFiles)));
+            @Auth MemberLoginInfo memberLoginInfo,
+            @RequestPart("productRequest") ProductUpdateRequestDTO productUpdateRequestDTO,
+            @RequestPart("imageFiles") List<MultipartFile> imageFiles) throws IOException {
+        return ResponseEntity.ok().body(new BaseResponse<>(
+                productService.updateProduct(productId, memberLoginInfo, productUpdateRequestDTO, imageFiles)));
     }
 
     /**
@@ -98,8 +96,8 @@ public class ProductController {
      */
     @DeleteMapping("/{productId}")
     public ResponseEntity<BaseResponse<String>> deleteProductById(@PathVariable("productId") Long productId,
-                                                                  @Auth MemberLoginInfo memberLoginInfo) {
-        return ResponseEntity.ok().body(new BaseResponse<>(productService.deleteById(memberLoginInfo, productId)));
+            @Auth MemberLoginInfo memberLoginInfo) {
+        return ResponseEntity.ok().body(new BaseResponse<>(productService.deleteById(productId)));
     }
 
 }

--- a/rebook-product/src/main/java/com/be/rebook/product/service/ProductService.java
+++ b/rebook-product/src/main/java/com/be/rebook/product/service/ProductService.java
@@ -37,11 +37,10 @@ public class ProductService {
     private final ProductRepository productRepository;
     private final S3Service s3Service;
 
-
     @Transactional
     public Long createProduct(MemberLoginInfo memberLoginInfo,
-                              ProductRequestDTO.ProductSaveRequestDTO productSaveRequestDTO,
-                              List<MultipartFile> imageFiles) throws IOException {
+            ProductRequestDTO.ProductSaveRequestDTO productSaveRequestDTO,
+            List<MultipartFile> imageFiles) throws IOException {
 
         if (memberLoginInfo == null) {
             log.error("Unauthorized access attempt - memberLoginInfo is null");
@@ -72,7 +71,8 @@ public class ProductService {
     /**
      * 상품 목록 조회
      */
-    public PaginationResponseDTO<ProductListResponseDTO> findAllProductByFilter(ProductRequestDTO.ProductFilterDTO productFilterDTO) {
+    public PaginationResponseDTO<ProductListResponseDTO> findAllProductByFilter(
+            ProductRequestDTO.ProductFilterDTO productFilterDTO) {
 
         log.info("dto 까보기");
         log.info("University: {}, Title: {}, Major: {}, MinPrice: {}, MaxPrice: {}",
@@ -85,8 +85,7 @@ public class ProductService {
         Pageable pageable = PageRequest.of(productFilterDTO.getPage(), productFilterDTO.getSize());
 
         Page<ProductListResponseDTO> productPage = productRepository.findProductsByFilter(productFilterDTO, pageable)
-                                                                 .map(ProductListResponseDTO::new);
-
+                .map(ProductListResponseDTO::new);
 
         return new PaginationResponseDTO<>(productPage);
     }
@@ -94,13 +93,9 @@ public class ProductService {
     /**
      * 내가 쓴 글 조회
      */
-    public List<ProductListResponseDTO> findAllMyProducts(MemberLoginInfo memberLoginInfo) {
-        if (memberLoginInfo == null) {
-            log.error("Unauthorized access attempt - memberLoginInfo is null");
-            throw new BaseException(ErrorCode.UNAUTHORIZED);
-        }
+    public List<ProductListResponseDTO> findAllMyProducts(String username) {
 
-        List<Product> products = productRepository.findProductsBySellerUsername(memberLoginInfo.getUsername());
+        List<Product> products = productRepository.findProductsBySellerUsername(username);
 
         productRepository.findById(1L);
         return products.stream()
@@ -123,9 +118,9 @@ public class ProductService {
      */
     @Transactional
     public Long updateProduct(Long productId,
-                              MemberLoginInfo memberLoginInfo,
-                              ProductRequestDTO.ProductUpdateRequestDTO productUpdateRequestDTO,
-                              List<MultipartFile> imageFiles) throws IOException {
+            MemberLoginInfo memberLoginInfo,
+            ProductRequestDTO.ProductUpdateRequestDTO productUpdateRequestDTO,
+            List<MultipartFile> imageFiles) throws IOException {
 
         if (memberLoginInfo == null) {
             log.error("Unauthorized access attempt - memberLoginInfo is null");
@@ -181,12 +176,7 @@ public class ProductService {
      * 상품 상태 변경
      */
     @Transactional
-    public Long changeStatus(MemberLoginInfo memberLoginInfo, Long productId, String status) {
-        if (memberLoginInfo == null) {
-            log.error("Unauthorized access attempt - memberLoginInfo is null");
-            throw new BaseException(ErrorCode.UNAUTHORIZED);
-        }
-
+    public Long changeStatus(Long productId, String status) {
         Product product = productRepository.findById(productId)
                 .orElseThrow(() -> new BaseException(ErrorCode.NOT_EXIST_PRODUCT));
 
@@ -198,20 +188,14 @@ public class ProductService {
      * 상품 삭제
      */
     @Transactional
-    public String deleteById(MemberLoginInfo memberLoginInfo, Long productId) {
-
-        if (memberLoginInfo == null) {
-            log.error("Unauthorized access attempt - memberLoginInfo is null");
-            throw new BaseException(ErrorCode.UNAUTHORIZED);
-        }
-
+    public String deleteById(Long productId) {
         Product product = productRepository.findById(productId)
-                        .orElseThrow(() -> new BaseException(ErrorCode.NOT_EXIST_PRODUCT));
+                .orElseThrow(() -> new BaseException(ErrorCode.NOT_EXIST_PRODUCT));
 
         List<String> fileNames = product.getProductImages().stream()
-                                        .map(ProductImage::getStoreFileName)
-                                        .collect(Collectors.toList());
-        
+                .map(ProductImage::getStoreFileName)
+                .collect(Collectors.toList());
+
         productRepository.deleteById(productId);
 
         fileNames.forEach(fileName -> {

